src/Renderer/RendererOpenGL.cpp

```diff
@@ -58,6 +58,13 @@ namespace {
 
 	GLfloat vertexArray[12] = {}; /**< Vertex array for quad drawing functions (all blitter functions). */
 	GLfloat textureCoordArray[12] = {}; /**< Texture coordinate array for quad drawing functions (all blitter functions). */
+
+
+	std::string glString(GLenum name)
+	{
+		const auto apiResult = glGetString(name);
+		return apiResult ? reinterpret_cast<const char*>(apiResult) : "";
```

Comment:
Consider using braced initialization and moving the cast wrap the `glGetString` call:
```
const auto* apiResult = reinterpret_cast<const char*>(glGetString(name));
return { apiResult ? apiResult : "" };
```

Reply:
That looks fine. I'm curious though, is there a particular reason?

Reply:
RVO gets invoked instead of a construct-then-copy/move. Makes it so there's no move and no copy in the first place.

Reply:
I'm not quite sure I follow. It should be the same either way? The `const char*` is used to construct a `std::string`. As that `std::string` is being directly returned RVO should apply.

Reply:
Copy elision versus RVO. six of one, half-a-dozen of the other. :P

Reply:
Add obligatory reference to [copy elision](https://en.cppreference.com/w/cpp/language/copy_elision) (which also talks about RVO and NRVO).

Reply:
So where do we stand on this now? The suggestions seem largely about style. I can't think of any technical reason for the change.

----

I think I would prefer leaving the cast where it is. It makes more sense to me to catch the raw API type/value as is. In some cases (not applicable here) if you were to interact further with an API, you would want values still in the original data type. Converting data (even if there is no actual runtime cost) seems like it should be done only when absolutely necessary.

----

The braces I could go either way on. They don't seem to do anything, so I'm a bit inclined to leave them off, though you could potentially make a case for readability or style here. Certainly they would be used when constructing `struct` return values.

Reply:
All I ask is the optioned be considered. Having done that...